core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -129,6 +129,42 @@ public void registerAsDeprecatedHandler(RestRequest.Method method, String path,
     }
 
     /**
+     * Registers a REST handler to be executed when the provided {@code method} and {@code path} match the request, or when provided
+     * with {@code deprecatedMethod} and {@code deprecatedPath}. Expected usage:
+     * <pre><code>
+     * // remove deprecation in next major release
+     * controller.registerWithDeprecatedHandler(POST, "/_forcemerge", this,
+     *                                          POST, "/_optimize", deprecationLogger);
+     * controller.registerWithDeprecatedHandler(POST, "/{index}/_forcemerge", this,
+     *                                          POST, "/{index}/_optimize", deprecationLogger);
+     * </code></pre>
+     * <p>
+     * The registered REST handler ({@code method} with {@code path}) is a normal REST handler that is not deprecated that is
+     * replacing the deprecated REST handler ({@code deprecatedMethod} with {@code deprecatedPath}) using the <em>same</em>
+     * {@code handler}.
```

Comment:
The language in this sentence isn't clear, perhaps change "that is replacing" to "and it is replacing"